### PR TITLE
Regression nan fix

### LINKED ISSUE #204: Regression is giving nan loss
**Describe the bug**
Starting a regression training job is giving nan loss.

**To Reproduce**
1. Start a regression training
2. See following warning:
```
...\venv\lib\site-packages\torch\nn\modules\loss.py:446: UserWarning: Using a target size (torch.Size([1])) that is different to the input size (torch.Size([0])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  return F.mse_loss(input, target, reduction=self.reduction)
```
3. And then this:
```
Looping over training data:  50%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                                                               | 189/380 [00:09<00:06, 29.60it/s]
Half-Epoch Average Train loss :  0.0
Half-Epoch Average Train mse :  nan
```

**Expected behavior**
Nan error should not happen

**Screenshots**

**GaNDLF Version**
0.0.12

**Desktop (please complete the following information):**
N.A.

**Additional context**
Reported by Noir

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #204

## Proposed Changes

- fix for regression nan
- includes #203

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [x] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [x] If adding a submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation

### CODE REVIEW

**Review by nigmatov — APPROVED** [review]
issue with NaN output for loss is fixed
